### PR TITLE
Add diagnostics for relation cache size

### PR DESCRIPTION
**Pull request by sandersn**

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -114,6 +114,9 @@ namespace ts {
             getIdentifierCount: () => sum(host.getSourceFiles(), "identifierCount"),
             getSymbolCount: () => sum(host.getSourceFiles(), "symbolCount") + symbolCount,
             getTypeCount: () => typeCount,
+            getAssignabilityCacheSize: () => assignableRelation.size,
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: Thoughts on doing it this way?
```ts
getRelationCacheSizes(): {
   assignable: assignableRelation.size,
   identity: identityRelation.size,
   subtype: subtypeRelation.size
}
```

Comment (DanielRosenwasser) [reply src/compiler/checker.ts]: No love for comparablility I see.

Comment (sandersn) [reply src/compiler/checker.ts]: @RyanCavanaugh seems like a fine idea. do you want to reduce the number of calls when getting perf diagnostics, or reduce the API surface? (or both)

@DanielRosenwasser  lolololololol

Comment (RyanCavanaugh) [reply src/compiler/checker.ts]: Yeah, both (seems like you'd rarely want only one of them and it's basically free to query)

**Pull request merged**